plugins/UM3NetworkPrinting/src/Cloud/Models/CloudClusterPrintJobStatus.py

```diff
@@ -77,11 +102,28 @@ def _createConfigurationModel(self) -> ConfigurationModel:
     ## Updates an UM3 print job output model based on this cloud cluster print job.
     #  \param model: The model to update.
     def updateOutputModel(self, model: UM3PrintJobOutputModel) -> None:
-        # TODO: Add `compatible_machine_families` to the cloud, than add model.setCompatibleMachineFamilies()
-        # TODO: Add `impediments_to_printing` to the cloud, see ClusterUM3OutputDevice._updatePrintJob
-        # TODO: Use model.updateConfigurationChanges, see ClusterUM3OutputDevice#_createConfigurationChanges
         model.updateConfiguration(self._createConfigurationModel())
         model.updateTimeTotal(self.time_total)
         model.updateTimeElapsed(self.time_elapsed)
         model.updateOwner(self.owner)
         model.updateState(self.status)
+        model.setCompatibleMachineFamilies(self.compatible_machine_families)
+        model.updateTimeTotal(self.time_total)
+        model.updateTimeElapsed(self.time_elapsed)
+        model.updateOwner(self.owner)
+
+        status_set_by_impediment = False
+        for impediment in self.impediments_to_printing:
+            if impediment.severity == "UNFIXABLE":  # TODO: impediment.severity is defined as int, this will not work, is there a translation?
```

Comment:
Regarding the TODO: the ClusterUM3OutputDevice probably has logic for this?

Reply:
No it doesn't, I checked, it  expects a string